File: listbinders/src/main/java/com/liusbl/listbinders/BinderNotFoundException.java
package com.liusbl.listbinders;

import java.util.List;

/**
 * Exception thrown when not all view types are provided
 */
@SuppressWarnings("rawtypes")
class BinderNotFoundException extends RuntimeException {
    public BinderNotFoundException(
            int viewType,
            List<LayoutBinder> binderList
    ) {
        super("Binder not found"); // TODO add information about viewType and binderList
    }
}

File: listbinders/src/main/java/com/liusbl/listbinders/BinderNotFoundException.kt
package com.liusbl.listbinders

/**
 * Exception thrown when not all view types are provided
 */
class BinderNotFoundException(
    viewType: Int,
    binderList: List<LayoutBinder<*>>
) : RuntimeException(
    "A binder handling view type \"$viewType\" position was not found.\n" +
            "Currently the handled view types are: [${binderList.joinToString { it.viewType.name }}]"
)